netket/variational.py

```diff
@@ -94,21 +95,36 @@ def _init_json_log(self):
         self._json_out["Output"] = []
 
     def run(
-        self, output_prefix, n_iter, step_size=1, save_params_every=50, write_every=50
+        self,
+        output_prefix,
+        n_iter,
+        step_size=1,
+        save_params_every=50,
+        write_every=50,
+        show_progress=True,
+        show_progress_fmt=".4f",
     ):
         self._init_json_log()
 
-        for k in range(n_iter):
-            self.advance(step_size)
-
-            self._add_to_json_log(k)
-            if k % write_every == 0 or k == n_iter - 1:
-                if self._mynode == 0:
-                    with open(output_prefix + ".log", "w") as outfile:
-                        json.dump(self._json_out, outfile)
-            if k % save_params_every == 0 or k == n_iter - 1:
-                if self._mynode == 0:
-                    self.machine.save(output_prefix + ".wf")
+        with trange(n_iter, disable=not show_progress) as itr:
+            for k in itr:
+                self.advance(step_size)
+
+                self._add_to_json_log(k)
+                if k % write_every == 0 or k == n_iter - 1:
+                    if self._mynode == 0:
+                        with open(output_prefix + ".log", "w") as outfile:
+                            json.dump(self._json_out, outfile)
+                if k % save_params_every == 0 or k == n_iter - 1:
+                    if self._mynode == 0:
+                        self.machine.save(output_prefix + ".wf")
+
+                # Update energy displayed in the progress bar
+                itr.set_postfix(
+                    Energy=("{0:" + show_progress_fmt + "}").format(
+                        self._vmc.energy_stats.mean.real
```

Comment:
Instead of just the energy mean, I'd prefer if you would display `str(stats)`, which looks like `<mean> ± <error> [var=<var>, R=<R>]`, as MC error, variance and also the `R` diagnostic are very helpful to detect problems and assess convergence during a run. (We should document what `R` means somewhere, by the way. Short version: Its supposed to be close to one if the MCMC chains are converged and larger when they are not.)

As `str(stats)` is also supposed to adjust the number of decimal places displayed according to the current MC error, this further allows you to remove the additional `show_progress_fmt` argument, which I consider a good think in order to keep the interface tidy.